components/generate_embeddings/src/main.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+import pandas as pd
+from fondant.component import PandasTransformComponent
+from langchain.embeddings import (
+    AlephAlphaAsymmetricSemanticEmbedding,
+    CohereEmbeddings,
+    HuggingFaceEmbeddings,
+    OpenAIEmbeddings,
+)
+from retry import retry
+from utils import to_env_vars
+
+logger = logging.getLogger(__name__)
+
+
+class GenerateEmbeddingsComponent(PandasTransformComponent):
+    def __init__(
+        self,
+        *_,
+        model_provider: str,
+        model: str,
+        api_keys: dict,
+    ):
+        self.model_provider = model_provider
+        self.model = model
+
+        to_env_vars(api_keys)
+
+    def get_embedding_model(self, model_provider, model: str):
+        # contains a first selection of embedding models
+        if model_provider == "aleph_alpha":
+            return AlephAlphaAsymmetricSemanticEmbedding(model=model)
+        if model_provider == "cohere":
+            return CohereEmbeddings(model=model)
+        if model_provider == "huggingface":
+            return HuggingFaceEmbeddings(model_name=model)
+        if model_provider == "openai":
+            return OpenAIEmbeddings(model=model)
+        msg = f"Unknown provider {model_provider}"
+        raise ValueError(msg)
+
+    @retry()  # make sure to keep trying even when api call limit is reached
```

Comment:
Does this keep retrying infinitely? Would be nice if this can be somehow be modified to log when a connection limit is reached.